dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/ext/MessageContextImpl.java

```diff
@@ -0,0 +1,322 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.jaxrs.ext;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Providers;
+
+import org.apache.cxf.attachment.AttachmentDeserializer;
+import org.apache.cxf.attachment.AttachmentImpl;
+import org.apache.cxf.attachment.AttachmentUtil;
+import org.apache.cxf.attachment.HeaderSizeExceededException;
+import org.apache.cxf.endpoint.Endpoint;
+import org.apache.cxf.helpers.CastUtils;
+import org.apache.cxf.interceptor.AttachmentOutInterceptor;
+import org.apache.cxf.io.CacheSizeExceededException;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.MultipartBody;
+import org.apache.cxf.jaxrs.impl.MetadataMap;
+import org.apache.cxf.jaxrs.impl.ProvidersImpl;
+import org.apache.cxf.jaxrs.interceptor.AttachmentInputInterceptor;
+import org.apache.cxf.jaxrs.interceptor.AttachmentOutputInterceptor;
+import org.apache.cxf.jaxrs.utils.ExceptionUtils;
+import org.apache.cxf.jaxrs.utils.JAXRSUtils;
+import org.apache.cxf.message.Exchange;
+import org.apache.cxf.message.ExchangeImpl;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.message.MessageImpl;
+import org.apache.cxf.message.MessageUtils;
+
+public class MessageContextImpl implements MessageContext {
+
+    private Message m;
+    public MessageContextImpl(Message m) {
+        this.m = m;
+    }
+
+    public Object get(Object key) {
+        String keyValue = key.toString();
+        if (MultipartBody.INBOUND_MESSAGE_ATTACHMENTS.equals(keyValue)
+            || (MultipartBody.INBOUND_MESSAGE_ATTACHMENTS + ".embedded").equals(keyValue)) {
+            try {
+                return createAttachments(key.toString());
+            } catch (CacheSizeExceededException e) {
+                m.getExchange().put("cxf.io.cacheinput", Boolean.FALSE);
+                throw new WebApplicationException(e, 413);
+            } catch (HeaderSizeExceededException e) {
+                throw new WebApplicationException(e, 413);
+            }
+        }
+        if (keyValue.equals("WRITE-" + Message.ATTACHMENTS)) {
+            // Liberty Change Start - #5049
+            if (m.getExchange().getOutMessage() == null) {
+                return null;
```

Comment:
Hmm... if we return null here, it seems like we might be preventing this method from returning something that might be on the "in" message from the code below - what if we did something like:
```
if (keyValue.equals("WRITE-" + Message.ATTACHMENTS) && m.getExchange().getOutMessage() != null) {
    return m.getExchange().getOutMessage().get(Message.ATTACHMENTS);
```